docs/source/primitives.rst

```diff
@@ -64,20 +68,20 @@ A flow can be just as list of tasks to be performed, a :ref:`while_loop_task` or
 While loop
 ----------
 
-Repeat the flow tasks until a condition fails, the condition is a set of validators once one of these validators fails the loop is broken and the result is inserted into the context above.
-
+Repeat the flow tasks until a condition fails. The condition is a set of validators, when any of these validators fail the loop will continue its current iteration but will not restart, the loop is in effect broken and the result is inserted into the context above.
+ASK_MICKY (probably need to reword this, but just trying to make it clear that it doesn't break immediately, it follows the normal pattern for a while and doesn't restart - maybe we should mention the concept of a break event here to illustrate how to get out immediately)
```

Comment:
note - validators are only checked at the start of a loop, so the loop will not enter the NEXT iteration